packages/arcgis-rest-request/src/job.ts

```diff
@@ -0,0 +1,494 @@
+  }
+
+  /**
+   * A handler that listens for a event once and returns a custom handler.
+   * 
+   * @param eventName A string of what event to listen for.
+   * @param handler A function of what to do when eventName was called.
+   */
+  once(eventName: string, handler: (e: any) => void) {
+    const fn = (arg: any) => {
+      this.emitter.off(eventName, fn);
+      handler(arg);
+    };
+
+    this.emitter.on(eventName, fn);
+
+    (handler as any).__arcgis_job_once_original_function__ = fn;
+  }
+
+  /**
+   * A handler that will remove a listener after its emitted and returns a custom handler. 
+   * 
+   * @param eventName A string of what event to listen for.
+   * @param handler A function of what to do when eventName was called.
+   */
+  off(eventName: string, handler: (e: any) => void) {
+    if ((handler as any).__arcgis_job_once_original_function__) {
+      this.emitter.off(
+        eventName,
+        (handler as any).__arcgis_job_once_original_function__
+      );
+      return;
+    }
+    this.emitter.off(eventName, handler);
+  }
+
+  /**
+   * Get the specific results of a successful job by result name. To get all results see {@linkcode Job.getAllResults}.
+   *
+   * If monitoring is disabled it will be enabled until the job classes resolves or rejects this promise.
+   *
+   * ```
+   * Job.submitJob(options)
+   *  .then((job) => {
+   *    return job.getResult("result_name")
+   *  }).then(result => {
+   *     console.log(result);
+   *   })
+   * ```
+   *
+   * @param result The name of the result that you want to retrieve.
+   * @returns An object representing the individual result of the job.
+   */
+  async getResult(result: string) {
+    return this.waitForJobCompletion().then((jobInfo) => {
+      return request(this.jobUrl + "/" + jobInfo.results[result].paramUrl, {
+        authentication: this.authentication
+      });
+    });
+  }
+
+  /**
+   * Formats the requestOptions to JSON format.
+   * 
+   * @returns An object as a JSON.
+   */
+  toJSON() {
+    return {
+      id: this.id,
+      url: this.url,
+      startMonitoring: this.isMonitoring,
+      pollingRate: this.pollingRate
+    };
+  }
+
+  /**
+   * Converts JSON to a single string.
+   * 
+   * @returns A string.
+   */
+  serialize() {
+    return JSON.stringify(this);
+  }
+
+  /**
+   * Checks for job status and if the job status is successful it resolves the job information, else if the jobStatus returns a failure state such as  "esriJobTimeOut", "esriJobCancelled", "esriJobFailed"
+   * it will reject the job information.
+   * 
+   * If neither of the above are true, this will return a new Promise and start {@linkcode Job.startInternalEventMonitoring} which will return a jobStatus and will only resolve the results if the job status comes 
+   * back as successful. All other status will be rejected and {@linkcode Job.stopInternalEventMonitoring} will be called.
+   * 
+   * @returns 
+   */
+  async waitForJobCompletion() {
```

Comment:
Another good place for the `IJobInfo` interface once it gets filled out.

```suggestion
  async waitForJobCompletion() : Promise<IJobInfo> {
```